src/Image.cc

```diff
@@ -10,6 +10,11 @@
 #include <node_buffer.h>
 #include "Util.h"
 
+/* Cairo limit:
+  * https://lists.cairographics.org/archives/cairo/2010-December/021422.html
+  */
+static constexpr const size_t canvas_max_side = (1 << 15) - 1;
```

Comment:
Don't need `const` with `constexpr` (`constexpr` is enough).

This can also definitely be an `int` instead of `size_t`. Anywhere it's compared against a larger type, it'll get implicitly promoted. (Think the only case of that is when reading a jpeg.)

(I'm only dwelling on the dimension types out of interest of minimizing the number of conversions.)